sample_pages/esp/eids-esp.1.0.js

```diff
@@ -0,0 +1,39 @@
+function fetchAsyncSignals(mode, customFunction, customKey) {
+	console.log("Going to fetch signals");
+	var eids = "";
+	switch (mode) {
+		case 1:
+			if (pbjs && pbjs.getUserIdsAsEids && typeof pbjs.getUserIdsAsEids === "function") {
+				eids = pbjs.getUserIdsAsEids(); // Get Identities from Prebid API in oRTB eids structure
+			}
+			break;
+		case 2:
+			if (owpbjs && owpbjs.getUserIdsAsEids && typeof owpbjs.getUserIdsAsEids === "function") {
+				eids = owpbjs.getUserIdsAsEids(); //Get Identities from Identity Hub  API in oRTB eids structure
+			}
+			break;
+		case 3:
+			if (typeof customFunction === "function") {
+				eids = customFunction(); //Get Identities from Custom function provided
+			}
+			break;
+
+		default:
+			eids = "Hello eids"; // Demo data
+
+	}
+	var dataKey = "eids" // Default key is eids
+	if (customKey && customKey.length > 0) {
+		dataKey = customKey; // Using custom key for data if passed  
+	}
+	var rawSignal = {}
+	rawSignal[dataKey] = eids;
+	var signals = encryptSignals(JSON.stringify(rawSignal))
+	promise = Promise.resolve(signals);
+	console.log("fetching Signals: " + signals);
+	return promise;
```

Comment:
![Codacy](https://codacy.pubmatic.com:443/assets/images/favicon.png) Issue found: ['promise' is not defined.](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15858)